### PR TITLE
refactor: Improve extension installation process

### PR DESCRIPTION
**Pull request by dominikriemer**
### Purpose

Refactor the installation process of extensions by better making use of the internal service discovery.
Improve the extension installation components in the UI.
Remove the support to register external endpoints.

### Remarks

PR introduces (a) breaking change(s): no

PR introduces (a) deprecation(s): yes
* `ExtensionsModelSubmitter` is deprecated. Extensions should directly inherit from `StreamPipesExtensionsServiceBase`.

### CODE REVIEW

File: streampipes-pipeline-management/src/main/java/org/apache/streampipes/manager/api/extensions/IExtensionsProvider.java [hunk 1]
```diff
@@ -16,17 +16,13 @@
...
 
-package org.apache.streampipes.storage.api;
+package org.apache.streampipes.manager.api.extensions;
 
-import org.apache.streampipes.model.client.endpoint.ExtensionsServiceEndpoint;
+import org.apache.streampipes.model.extensions.ExtensionItemDescription;
 
 import java.util.List;
 
-public interface IExtensionsServiceEndpointStorage {
```
Comment (tenthe) [inline streampipes-pipeline-management/src/main/java/org/apache/streampipes/manager/api/extensions/IExtensionsProvider.java:25]: This interface seems not to be used anymore, can  it be deleted?

File: streampipes-pipeline-management/src/main/java/org/apache/streampipes/manager/endpoint/AvailableExtensionsProvider.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+public class AvailableExtensionsProvider {
+
+  private final INoSqlStorage storage;
+
+  public AvailableExtensionsProvider(INoSqlStorage storage) {
+    this.storage = storage;
+  }
+
+  public List<ExtensionItemDescription> getExtensions() {
```
Comment (tenthe) [inline streampipes-pipeline-management/src/main/java/org/apache/streampipes/manager/endpoint/AvailableExtensionsProvider.java:43]: Can you provide a few comments what is going on in this method? For me it is not quite clear what the logic is to geht all ExtensionItemDescriptions.

File: streampipes-pipeline-management/src/main/java/org/apache/streampipes/manager/endpoint/AvailableExtensionsProvider.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+import java.util.stream.Collectors;
+
+public class AvailableExtensionsProvider {
+
+  private final INoSqlStorage storage;
+
+  public AvailableExtensionsProvider(INoSqlStorage storage) {
+    this.storage = storage;
+  }
+
```
Comment (tenthe) [inline streampipes-pipeline-management/src/main/java/org/apache/streampipes/manager/endpoint/AvailableExtensionsProvider.java:42]: Maybe the method can also be renamed to `getExtensionItemDescriptions` or `getExtensionItems`. What do you think?

File: ui/src/app/add/components/endpoint-item/endpoint-item.component.ts [hunk 4]
```diff
@@ -47,12 +48,19 @@ export class EndpointItemComponent implements OnInit {
     iconReady = false;
     iconError = false;
 
+    cssMapper: Record<string, string> = {
+        ADAPTER: 'adapter',
+        DATA_STREAM: 'stream',
+        DATA_PROCESSOR: 'sepa',
```
Comment (tenthe) [inline ui/src/app/add/components/endpoint-item/endpoint-item.component.ts:54]: Is it possible to replace sepa with processor? If the changes would be too big we should create a PR for this task

**Review by tenthe — CHANGES_REQUESTED** [review]
Thanks for the PR. I have added some code reviews.
If something is not clear, please let me know.

File: streampipes-model/src/main/java/org/apache/streampipes/model/extensions/ExtensionAsset.java [hunk 5]
```diff
@@ -15,15 +15,11 @@
...
  *
  */
-package org.apache.streampipes.svcdiscovery.api.model;
 
-import org.apache.streampipes.model.extensions.svcdiscovery.SpServiceTag;
-import org.apache.streampipes.model.extensions.svcdiscovery.SpServiceTagPrefix;
+package org.apache.streampipes.model.extensions;
 
-public class DefaultSpServiceTags {
+public class ExtensionAsset {
```
[the remark this answers is not in the document]
Comment (dominikriemer) [reply streampipes-model/src/main/java/org/apache/streampipes/model/extensions/ExtensionAsset.java]: done

File: streampipes-service-extensions/src/main/java/org/apache/streampipes/service/extensions/StreamPipesExtensionsServiceBase.java [hunk 6]
```diff
@@ -98,22 +126,50 @@ private void registerService(SpServiceRegistration serviceRegistration) {
...
+        )
+        .peek(entity -> {
+          try {
+            if (entity.isIncludesLocales()) {
+              var labelGenerator = new LabelGenerator<>(entity);
+              entity.setName(labelGenerator.getElementTitle());
+              entity.setDescription(labelGenerator.getElementDescription());
+            }
+          } catch (IOException e) {
+
```
[the remark this answers is not in the document]
Comment (dominikriemer) [reply streampipes-service-extensions/src/main/java/org/apache/streampipes/service/extensions/StreamPipesExtensionsServiceBase.java]: done

File: streampipes-service-extensions/src/main/java/org/apache/streampipes/service/extensions/StreamPipesExtensionsServiceBase.java [hunk 7]
```diff
@@ -98,22 +126,50 @@ private void registerService(SpServiceRegistration serviceRegistration) {
...
     if (DeclarersSingleton.getInstance().getServiceDefinition() != null) {
       tags.add(SpServiceTag.create(SpServiceTagPrefix.SP_GROUP,
           DeclarersSingleton.getInstance().getServiceDefinition().getServiceGroup()));
     }
-    tags.addAll(getExtensionsServiceTags());
+    tags.addAll(getExtensionsServiceTags(extensions));
     return tags;
   }
 
+  private Set<ExtensionItemDescription> getAllExtensions() {
```
[the remark this answers is not in the document]
Comment (dominikriemer) [reply streampipes-service-extensions/src/main/java/org/apache/streampipes/service/extensions/StreamPipesExtensionsServiceBase.java]: I created a separate class to provide extensions

File: streampipes-pipeline-management/src/main/java/org/apache/streampipes/manager/api/extensions/IExtensionsProvider.java (hunk 1 above)
Comment (dominikriemer) [reply streampipes-pipeline-management/src/main/java/org/apache/streampipes/manager/api/extensions/IExtensionsProvider.java]: deleted

File: streampipes-pipeline-management/src/main/java/org/apache/streampipes/manager/endpoint/AvailableExtensionsProvider.java (hunk 2 above)
Comment (dominikriemer) [reply streampipes-pipeline-management/src/main/java/org/apache/streampipes/manager/endpoint/AvailableExtensionsProvider.java]: added comments

File: streampipes-pipeline-management/src/main/java/org/apache/streampipes/manager/endpoint/AvailableExtensionsProvider.java (hunk 3 above)
Comment (dominikriemer) [reply streampipes-pipeline-management/src/main/java/org/apache/streampipes/manager/endpoint/AvailableExtensionsProvider.java]: Renamed the method and split the logic into smaller chunks

File: ui/src/app/add/components/endpoint-item/endpoint-item.component.ts (hunk 4 above)
Comment (dominikriemer) [reply ui/src/app/add/components/endpoint-item/endpoint-item.component.ts:54]: That's a larger undertaking we can address in a separate PR

**Pull request merged**